### DISCUSSION TITLE
Absolute path import does not work for .storybook/main.ts

### DISCUSSION THREAD

**Discussion by Lee-Si-Yoon** [storybookjs/storybook#25028] [Help] [answered]
### Summary

Why does absolute imports do not work for `storybook/main.ts` only?
In a monorepo I have structure like this

### File structure
```
├ App1
│  └ .storybook
│    └ main.ts
│    └ preview.ts
│  └ src
│    └ *.stories.tsx
│ └ tsconfig.json
│
├ App2
│  └ .storybook
│    └ main.ts
│    └ preview.ts
│  └ src
│    └ *.stories.tsx
│ └ tsconfig.json
│
├ Libs
│  └ .storybook
│    └ main.ts
│    └ preview.ts
│  └ src
│    └ SharedDecorator.tsx
│ └ tsconfig.json
│
├ tsconfig.base.json
```

for instance in App1's `preview.ts` absolute path import for `import { A } from "@Libs/SharedDecorator"` works fine
but for App1's `main.ts` absolute path import for `import { A } from "@Libs/SharedDecorator"` returns module not found

I want to know what is causing this difference

**Comment by chantastic** [comment]
can you share the tsconfig.json files for both?

**Reply by Lee-Si-Yoon** [reply]
of course

in the root(tsconfig.base.json) 

```json
{
  "compilerOptions": {
    "rootDir": ".",
    "baseUrl": ".",
    "paths": {
      "@Libs/*": ["Libs/*"],
    },
  "exclude": ["node_modules", "tmp"]
}
```

App1's tsconfig is

```json
{
  "extends": "../tsconfig.base.json",
  "compilerOptions": {
    "types": ["node", "jest", "vite/client"],
    "strict": true,
    "noEmit": true
  },
  "include": [
    "**/*.js",
    "**/*.jsx",
    "**/*.ts",
    "**/*.tsx",
    "**/*.d.ts",
    ".storybook/*.ts",
    "**/*.stories.ts",
    "**/*.stories.tsx",
    "**/*.stories.mdx"
  ],
  "exclude": ["node_modules"]
}
```

and for the Libs is identical to App1's one

**Comment by valentinpalkovic** [comment, marked as answer]
This is a duplicate of https://github.com/storybookjs/storybook/issues/22078. See also discussion in https://github.com/storybookjs/storybook/pull/24765
